src/plugins/intel_cpu/src/transformations/tpp/x64/pass/fuse_tpp_to_equations.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "fuse_tpp_to_equations.hpp"
+
+#include "transformations/tpp/x64/op/eltwise.hpp"
+#include "transformations/tpp/x64/op/equation.hpp"
+#include "snippets/utils/utils.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/lowered/port_descriptor.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace pass {
+using snippets::lowered::ExpressionPtr;
+using snippets::lowered::ExpressionPort;
+using NodePtr = std::shared_ptr<Node>;
+
+bool FuseTPPToEquations::fuse_from_root(const NodePtr& root, const std::shared_ptr<ov::Model>& m) {
+    using snippets::lowered::PortDescriptorUtils;
+    OutputVector eq_ivals;
+    std::vector<op::OpDescTPP> op_descs;
+    std::unordered_map<NodePtr , NodePtr> node_replace_map;
+    // Only ops with one out are supported due to Equations restrictions
+    auto supported_num_out = [](const Output<ov::Node>& out) {
+        const auto& n = out.get_node_shared_ptr();
+        return n->get_output_size() == 1 && out.get_target_inputs().size() == 1;
+    };
+    auto get_tpp_op = [](const NodePtr& n) {
+        auto tpp = std::dynamic_pointer_cast<op::EltwiseTPP>(n);
+        bool not_supported_op =
+                // ticket: 152532
+                ov::is_type<ov::snippets::op::ReduceBase>(n) ||
+                // ticket: 152510
+                ov::is_type<ov::op::v0::Relu>(n);
+        return not_supported_op ? nullptr : tpp;
+    };
+
+    // Note: we don't support exprs with more than 1 output yet. It's a technical limitation, but there are no use cases
+    const auto tpp_root = get_tpp_op(root);
+    if (!tpp_root || !supported_num_out(root->output(0)))
+        return false;
+
+    const auto root_subtensor = PortDescriptorUtils::get_port_descriptor_ptr(root->output(0))->get_subtensor();
+    auto supported_subtensor = [&root_subtensor](const snippets::VectorDims& subtensor) {
+        const auto size = subtensor.size();
+        if (size != root_subtensor.size())
+            return false;
+        for (size_t i = 0; i < size; i++) {
+            if (subtensor[i] != root_subtensor[i] && subtensor[i] != 1)
+                return false;
+        }
+        return true;
+    };
+
+    std::vector<op::OpDescTPP> op_types;
```

Comment:
`op_types` is not used, please remove it

Reply:
Removed, thanks